dcp_prototype/frontend/src/components/login-signup.js

```diff
@@ -0,0 +1,19 @@
+import React from "react"
+import { useAuth0 } from "../contexts/auth0Context"
+import { Heading } from "theme-ui"
+
+const LoginSignup = () => {
+  const { isAuthenticated, loading, loginWithRedirect } = useAuth0()
```

Comment:
Wondering if there's a pattern we can use that initializes `auth0` when necessary (e.g. per page as opposed to per component). Just something to think about. OK with this long term if the overhead of calling `useAuth0()` is minimal.

Reply:
From my limit react and gatsby knowledge, it is initialized per page. Using the `wrapRootElement` in _gatsby-browser.js_ and _gatsby-node.js_ we wrap every page in an Auth0 context that initializes Auth0 for every page. The `loading` flag is used to determine if the async response from Auth0 has completed.